x-pack/plugins/security_solution/public/detections/components/rules/step_define_rule/use_previous.ts

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect, useRef } from 'react';
+
+export const usePrevious = <T>(value: T): T | undefined => {
```

Comment:
Instead of defining this again, you can also import this one: `import usePrevious from 'react-use/lib/usePrevious';`